types/label.d.ts

```diff
@@ -26,90 +75,51 @@ export interface LabelOptions {
-
-  /**
-	 * Border radius of the label rectangle
-	 * @default 6
-	 */
+   * Border radius of the label rectangle
+   * @default 6
+   */
   borderRadius?: Scriptable<number, PartialEventContext>,
-
   /**
    * @deprecated replaced by borderRadius
    * @todo remove at v2
    */
   cornerRadius?: Scriptable<number, PartialEventContext>,
+}
 
-  /**
-	 * Anchor position of label on line.
-	 * @default 'center'
-	 */
+export interface LabelOptions extends ContainedLabelOptions {
   position?: Scriptable<LabelPosition, PartialEventContext>,
-
   /**
-	 * Text alignment when the content of the label is multi-line.
-	 * @default 'center'
-	 */
-  textAlign?: Scriptable<LabelTextAlign, PartialEventContext>,
-
-  /**
-	 * Adjustment along x-axis (left-right) of label relative to above number (can be negative)
-	 * For horizontal lines positioned left or right, negative values move
-	 * the label toward the edge, and positive values toward the center.
-	 * @default 0
-	 */
-  xAdjust?: Scriptable<number, PartialEventContext>,
-
+   * Whether the label is enabled and should be displayed
+   * @default true
+   */
+  enabled?: Scriptable<boolean, PartialEventContext>,
   /**
-	 * Adjustment along y-axis (top-bottom) of label relative to above number (can be negative)
-	 * For vertical lines positioned top or bottom, negative values move
-	 * the label toward the edge, and positive values toward the center.
-	 * @default 0
-	 */
-  yAdjust?: Scriptable<number, PartialEventContext>,
+   * Rotation of label, in degrees, or 'auto' to use the degrees of the line, default is 0
+   * @default 90
+   */
+  rotation?: Scriptable<number | 'auto', PartialEventContext>
+}
 
+export interface BoxLabelOptions extends CoreLabelOptions {
+  position?: Scriptable<LabelPosition | LabelPositionObject, PartialEventContext>,
   /**
-	 * Whether the label is enabled and should be displayed
-	 * @default true
-	 */
+   * Whether the label is enabled and should be displayed
+   * @default true
+   */
   enabled?: Scriptable<boolean, PartialEventContext>,
+}
 
-  /**
-	 * Text to display in label. Provide an array to display multiple lines
-	 */
-  content: Scriptable<string | string[] | HTMLImageElement | null, PartialEventContext>,
+export interface LabelTypeOptions extends ContainedLabelOptions {
+  position?: Scriptable<LabelPosition | LabelPositionObject, PartialEventContext>,
+}
 
-  /**
-	 * Overrides the width of the image. Could be set in pixel by a number,
-	 * or in percentage of current width of image by a string
-	 */
-  width?: Scriptable<number | string, PartialEventContext>,
+export type LabelPosition = 'start' | 'center' | 'end';
 
-  /**
-	 * Overrides the height of the image. Could be set in pixel by a number,
-	 * or in percentage of current height of image by a string
-	 */
-  height?: Scriptable<number | string, PartialEventContext>,
+export type LabelTextAlign = 'left' | 'start' | 'center' | 'right' | 'end';
```

Comment:
This was why I asked about the previous question. Need to have consistency. It's probably better to support also left/right too, for better UX (or is it DX in this case) .

Reply:
fixed